src/Test/QuickCheck/StateMachineModel.hs

```diff
@@ -304,16 +305,25 @@ linearise
   :: forall cmd ix model
   .  StateMachineModel model cmd ix
   -> History cmd ix
-  -> [[Operation cmd ix]]
-linearise StateMachineModel {..} = map fromRose . filter (postConditionsHold initialModel) . linearTree
+  -> Property
+linearise _                      [] = property True
+linearise StateMachineModel {..} xs = go initialModel . linearTree $ xs
   where
-  postConditionsHold :: model -> Rose (Operation cmd ix) -> Bool
-  postConditionsHold m (Rose (Operation cmd resp _ _) roses) =
-    postcondition m cmd resp && any' (postConditionsHold (transition m cmd resp)) roses
-    where
-    any' :: (a -> Bool) -> [a] -> Bool
-    any' _ [] = True
-    any' p xs = any p xs
+  go :: model -> [Rose (Operation cmd ix)] -> Property
+  go _ []       = property False
+  go m (o : os) = step m o .||. go m os
+
+  step :: model -> Rose (Operation cmd ix) -> Property
+  step m (Rose (Operation cmd resp _ _) roses) = postcondition m cmd resp .&&.
+    anyP (step (transition m cmd resp)) roses
+
+  anyP :: (a -> Property) -> [a] -> Property
```

Comment:
I think we should rename this one since `anyP'` is the "real" `anyP`. Something like `anyPorDone`?